tests/testthat/test-renderer2-widerect.R

```diff
@@ -1,454 +1,428 @@
-# This is kinda a hack that causes it to backspace the last element in the list
-afg.div$clickElement()
-remDr$sendKeysToActiveElement(list(key="backspace"))
-Sys.sleep(1)
-
-test_that("backspace removes Afghanistan from selected countries", {
-  country.vec <- getCountries()
-  expected.countries <- c("United States", "Vietnam")
-  expect_identical(sort(country.vec), sort(expected.countries))
-})
-
-getWidth <- function(){
-  node.set <-
-    getNodeSet(getHTML(), '//g[@class="geom10_bar_bar"]//rect[@id="Vietnam"]')
-  expect_equal(length(node.set), 1)
-  alist <- xmlAttrs(node.set[[1]])
-  alist[["width"]]
-}
-
-test_that("middle of transition != after when duration=2000", {
-  clickID("year1960")
-  Sys.sleep(1)
-  before.width <- getWidth()
-  clickID("year2010")
-  during.width <- getWidth()
-  Sys.sleep(0.1)
-  after.width <- getWidth()
-  rbind(before=before.width,
-        during=during.width,
-        after=after.width)
-  expect_true(during.width != after.width)
-})
-
-e <- remDr$findElement("id", "plot_duration_ms_year")
-e$clickElement()
-e$clearElement()
-e$sendKeysToElement(list("0", key="enter"))
-Sys.sleep(1)
-
-test_that("middle of transition == after when duration=0", {
-  clickID("year1960")
-  Sys.sleep(1)
-  before.width <- getWidth()
-  clickID("year2010")
-  during.width <- getWidth()
-  Sys.sleep(0.1)
-  after.width <- getWidth()
-  rbind(before=before.width,
-        during=during.width,
-        after=after.width)
-  expect_true(before.width != after.width)
-  expect_true(during.width == after.width)
-})
+acontext("geom_widerect")
+library(animint2)
+expect_source <- function(expected){
+  a.list <- getNodeSet(info$html, '//a[@id="a_source_href"]')
+  computed <- if(length(a.list)==0){
+    NULL
+  }else{
+    at.mat <- sapply(a.list, xmlAttrs)
+    at.mat["href",]
+  }
+  expect_identical(as.character(computed), as.character(expected))
+}
+
+recommendation <- data.frame(
+  min.C=21,
+  max.C=23)
+set.seed(1)
+temp.time <- data.frame(
+  time=strptime(paste0("2015-10-", 1:31), "%Y-%m-%d"),
+  temp.C=rnorm(31))
+viz <- animint(
+  gg=ggplot()+
+    theme_bw()+
+    theme_animint(height=200, width=2000)+
+    geom_widerect(aes(
+      ymin=min.C, ymax=max.C),
+      color=NA,
+      fill="grey",
+      data=recommendation)+
+    geom_line(aes(
+      time, temp.C),
+      data=temp.time)
+)
+
+info <- animint2HTML(viz)
+expect_source(NULL)
+
+getBounds <- function(geom.class){
+ 
+    script.txt <- sprintf('return document.getElementsByClassName("%s")[0].getBoundingClientRect()', geom.class)
+    remDr$executeScript(script.txt)
+}
+
+test_that("bottom of widerect is above line", {
+  
+  
+  rect_bound_script <- 'document.getElementsByClassName("geom1_widerect_gg")[0].getBoundingClientRect().bottom;'
+  rect_bound <- remDr$Runtime$evaluate(rect_bound_script,returnByValue = TRUE)$result$value
+  line_bound_script <- 'document.getElementsByClassName("geom2_line_gg")[0].getBoundingClientRect().top;'
```

Comment:
please edit getBounds and use it twice instead of repeating JS code here

Reply:
Edited and used getBounds() instead of JS code